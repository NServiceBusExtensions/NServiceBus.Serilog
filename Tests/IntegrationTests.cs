using System.Collections.Generic;
using System.Threading.Tasks;
using NServiceBus;
using NServiceBus.Logging;
using NServiceBus.Serilog;
using NUnit.Framework;
using Serilog;
using Serilog.Events;

[TestFixture]
[Explicit]
public class IntegrationTests
{
    [Test]
    public async Task Ensure_log_messages_are_redirected()
    {
        var logs = new List<LogEvent>();
        var eventSink = new EventSink
        {
            Action = logs.Add
        };

        Log.Logger = new LoggerConfiguration()
            .MinimumLevel.Verbose()
            .WriteTo.Sink(eventSink)
            .CreateLogger();
        LogManager.Use<SerilogFactory>();

        var endpointConfiguration = new EndpointConfiguration("SerilogTests");
        endpointConfiguration.UseSerialization<JsonSerializer>();
        endpointConfiguration.EnableInstallers();
        endpointConfiguration.SendFailedMessagesTo("error");
        endpointConfiguration.UsePersistence<InMemoryPersistence>();

        var endpoint = await Endpoint.Start(endpointConfiguration);
        try
        {
            Assert.IsNotEmpty(logs);
        }
        finally
        {
            await endpoint.Stop();
        }
    }
}