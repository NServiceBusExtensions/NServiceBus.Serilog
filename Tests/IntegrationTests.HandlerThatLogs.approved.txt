{
  "logsForTarget": [
    {
      "MessageTemplate": {
        "Text": "Receive message {MessageType} {MessageId}.",
        "Tokens": [
          {
            "Length": 16,
            "Text": "Receive message "
          },
          {
            "Length": 13,
            "PropertyName": "MessageType",
            "StartIndex": 16
          },
          {
            "Length": 1,
            "Text": " ",
            "StartIndex": 29
          },
          {
            "Length": 11,
            "PropertyName": "MessageId",
            "StartIndex": 30
          },
          {
            "Length": 1,
            "Text": ".",
            "StartIndex": 41
          }
        ]
      },
      "Level": "Information",
      "Properties": {
        "Message": {
          "TypeTag": "StartHandlerThatLogs",
          "Properties": []
        },
        "OriginatingHostId": {
          "Value": "Guid 1"
        },
        "MessageIntent": {
          "Value": "Send"
        },
        "OriginatingEndpoint": {
          "Value": "SerilogTests"
        },
        "OriginatingMachine": {
          "Value": "MachineName"
        },
        "ReplyToAddress": {
          "Value": "SerilogTests"
        },
        "MessageId": {
          "Value": "Guid 2"
        },
        "MessageType": {
          "Value": "StartHandlerThatLogs"
        },
        "CorrelationId": {
          "Value": "Guid 2"
        },
        "ConversationId": {
          "Value": "Guid 3"
        },
        "SourceContext": {
          "Value": "StartHandlerThatLogs"
        },
        "ProcessingEndpoint": {
          "Value": "SerilogTests"
        }
      }
    },
    {
      "MessageTemplate": {
        "Text": "Sent message {MessageType} {MessageId}.",
        "Tokens": [
          {
            "Length": 13,
            "Text": "Sent message "
          },
          {
            "Length": 13,
            "PropertyName": "MessageType",
            "StartIndex": 13
          },
          {
            "Length": 1,
            "Text": " ",
            "StartIndex": 26
          },
          {
            "Length": 11,
            "PropertyName": "MessageId",
            "StartIndex": 27
          },
          {
            "Length": 1,
            "Text": ".",
            "StartIndex": 38
          }
        ]
      },
      "Level": "Information",
      "Properties": {
        "Message": {
          "TypeTag": "StartHandlerThatLogs",
          "Properties": []
        },
        "UnicastRoutes": {
          "Elements": [
            {
              "Value": "SerilogTests"
            }
          ]
        },
        "OriginatingHostId": {
          "Value": "Guid 1"
        },
        "MessageIntent": {
          "Value": "Send"
        },
        "OriginatingEndpoint": {
          "Value": "SerilogTests"
        },
        "OriginatingMachine": {
          "Value": "MachineName"
        },
        "ReplyToAddress": {
          "Value": "SerilogTests"
        },
        "MessageId": {
          "Value": "Guid 2"
        },
        "MessageType": {
          "Value": "StartHandlerThatLogs"
        },
        "CorrelationId": {
          "Value": "Guid 2"
        },
        "ConversationId": {
          "Value": "Guid 3"
        },
        "SourceContext": {
          "Value": "StartHandlerThatLogs"
        },
        "ProcessingEndpoint": {
          "Value": "SerilogTests"
        }
      }
    }
  ],
  "logsWithExceptions": [
    {
      "MessageTemplate": {
        "Text": "The message",
        "Tokens": [
          {
            "Length": 11,
            "Text": "The message"
          }
        ]
      },
      "Level": "Error",
      "Properties": {
        "SourceContext": {
          "Value": "TheHandlerThatThrows"
        },
        "ExceptionDetail": {
          "Elements": {
            "\"Type\"": {
              "Value": "System.Exception"
            },
            "\"HResult\"": {
              "Value": -2146233088
            },
            "\"Message\"": {
              "Value": "Exception of type 'System.Exception' was thrown."
            },
            "\"Source\"": {}
          }
        }
      },
      "Exception": {
        "ClassName": "System.Exception",
        "Message": null,
        "Data": {},
        "InnerException": null,
        "HelpURL": null,
        "RemoteStackIndex": 0,
        "ExceptionMethod": null,
        "HResult": -2146233088,
        "Source": null,
        "WatsonBuckets": null
      }
    }
  ]
}
